deepdrr/vol.py

```diff
@@ -159,11 +160,132 @@ def from_nifti(
+    ):
+        """
+        load a volume from a dicom file and compute the anatomical_from_ijk transform from metadata
+        https://www.slicer.org/wiki/Coordinate_systems
+        Args:
+            path: path-like to a multi-frame dicom file. (Currently only Multi-Frame from Siemens supported)
+            use_thresholding (bool, optional): segment the materials using thresholding (faster but less accurate). Defaults to True.
+            world_from_anatomical (Optional[geo.FrameTransform], optional): position the volume in world space. If None, uses identity. Defaults to None.
+            use_cached (bool, optional): [description]. Use a cached segmentation if available. Defaults to True.
+            cache_dir (Optional[Path], optional): Where to load/save the cached segmentation. If None, use the parent dir of `path`. Defaults to None.
+
+        Returns:
+            Volume: an instance of a deepdrr volume
+        """
+        path = Path(path)
+        stem = path.name.split('.')[0]
+
+        if cache_dir is None:
+            cache_dir = path.parent
+
+        # Multi-frame dicoms store all slices of a volume in one file.
+        # they must specify the necessary dicom tags under
+        # https://dicom.innolitics.com/ciods/enhanced-ct-image/enhanced-ct-image-multi-frame-functional-groups
+        assert path.is_file(), 'Currently only multi-frame dicoms are supported. Path must refer to a file.'
+        logger.info(f'loading Dicom volume from {path}')
+
+        # reading the dicom dataset object
+        ds = dcmread(path)
+
+        # extracting all needed tags TODO add try exepts
+        frames = ds.PerFrameFunctionalGroupsSequence
+        shared = ds.SharedFunctionalGroupsSequence[0]
+        num_slices = len(frames)
+        first_slice_position = np.array(frames[0].PlanePositionSequence[0].ImagePositionPatient)
+        last_slice_position = np.array(frames[-1].PlanePositionSequence[0].ImagePositionPatient)
+        RC = np.array(shared.PlaneOrientationSequence[0].ImageOrientationPatient).reshape(2, 3).T
+        PixelSpacing = np.array(ds.SharedFunctionalGroupsSequence[0].PixelMeasuresSequence[0].PixelSpacing)
+        SliceThickness = np.array(ds.SharedFunctionalGroupsSequence[0].PixelMeasuresSequence[0].SliceThickness)
+        offset = ds.SharedFunctionalGroupsSequence[0].PixelValueTransformationSequence[0].RescaleIntercept
+        scale = ds.SharedFunctionalGroupsSequence[0].PixelValueTransformationSequence[0].RescaleSlope
+
+        # make user aware that this is only tested on windows
+        if ds.Manufacturer != "SIEMENS":
+            logger.warning("Multi-frame loading has only been tested on Siemens Enhanced CT DICOMs."
+                           "Please verify everything works as expected.")
+
+        # read the 'raw' data array
+        raw_data = ds.pixel_array.astype(np.float32)
+        hu_values = raw_data * scale + offset
+
+        # move slice index axis to back (k, j, i) -> (j, i, k)
```

Comment:
Directly change the data to `(i, j, k)` format, as the framework will adhere to this convention too.

Reply:
done